src/elements/common/content-answers/ContentAnswersModal.tsx

```diff
@@ -0,0 +1,199 @@
+import APIFactory from '../../../api';
+// @ts-ignore: no ts definition
+// eslint-disable-next-line import/named
+import { ElementsXhrError } from '../../common/types/api';
+
+import messages from './messages';
+
+export interface ExternalProps {
+    isCitationsEnabled?: boolean;
+    isMarkdownEnabled?: boolean;
+    isResetChatEnabled?: boolean;
+    onAsk?: () => void;
+    onClearConversation?: () => void;
+    onRequestClose?: () => void;
+    suggestedQuestions?: SuggestedQuestionType[];
+}
+
+export interface ContentAnswersModalProps extends ExternalProps {
+    api: APIFactory;
+    currentUser?: User;
+    file: BoxItem;
+    isOpen: boolean;
+}
+
+const ContentAnswersModal = ({
+    api,
+    currentUser,
+    file,
+    isOpen,
+    onAsk,
+    onClearConversation,
+    onRequestClose,
+    suggestedQuestions,
+    isCitationsEnabled = true,
+    isMarkdownEnabled = true,
+    isResetChatEnabled = true,
+}: ContentAnswersModalProps) => {
+    const { formatMessage } = useIntl();
+    const [isLoading, setIsLoading] = useState<boolean>(false);
+    const [questions, setQuestions] = useState<QuestionType[]>([]);
+    let localizedQuestions: SuggestedQuestionType[] = [];
+
+    if (!suggestedQuestions || suggestedQuestions.length === 0) {
+        localizedQuestions = DOCUMENT_SUGGESTED_QUESTIONS.map(question => ({
+            id: question.id,
+            label: formatMessage(messages[question.labelId]),
+            prompt: formatMessage(messages[question.promptId]),
+        }));
+    }
+
+    const handleSuccessCallback = useCallback((response: AxiosResponse): void => {
+        const question = {
+            answer: response.data.answer,
+            created_at: response.data.created_at,
+            citations: response.data.citations,
+            error: null,
+            isCompleted: true,
+            isLoading: false,
+        };
+
+        setQuestions(prevState => {
+            const lastQuestion = prevState[prevState.length - 1];
+            const updatedLastQuestion = { ...lastQuestion, ...question };
+            return [...prevState.slice(0, -1), updatedLastQuestion];
+        });
+    }, []);
+
+    const handleErrorCallback = useCallback((error: ElementsXhrError, question: QuestionType): void => {
+        const rateLimitingRegex = /Too Many Requests/i;
+
+        const errorMessage = error?.message || '';
+        const isRateLimitingError = error?.response?.status === 429 || rateLimitingRegex.test(errorMessage);
+
+        const errorQuestion = {
+            ...question,
+            error: isRateLimitingError ? ANSWER_ERROR.RATE_LIMITING : ANSWER_ERROR.GENERAL,
+            isCompleted: true,
+            isLoading: false,
+        };
+
+        setQuestions(prevState => {
+            return [...prevState.slice(0, -1), errorQuestion];
+        });
+    }, []);
+
+    const handleAsk = useCallback(
+        async (question: QuestionType, aiAgent: AgentType, isRetry = false) => {
```

Comment:
what's `aiAgent` for?

Reply:
optional option that is not present enabled publicly